coreline_ftco_pos_ex2/static/src/Screens/ProductScreen/ControlButtons/SelectBundleButton/SelectBundleButton.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+import PosComponent from "point_of_sale.PosComponent";
+import Registries from "point_of_sale.Registries";
+import ProductScreen from "point_of_sale.ProductScreen";
+
+export class SelectBundleButton extends PosComponent {
+    get currentOrder() {
+        return this.env.pos.get_order();
+    }
+
+    async onClick() {
+        const bundlesList = [];
+        for (let bundle of this.env.pos.bundles) {
+            bundlesList.push({
+                id: bundle.id,
+                label: bundle.name,
+                item: bundle,
+            });
+        }
```

Comment:
I would personally go for a functional approach here to make it easier to maintain


```js
const bundlesList = this.env.pos.bundles.map(bundle => ({
    id: bundle.id,
    label: bundle.name,
    item: bundle
}));
```